woocommerce/woocommerce-gutenberg-products-block
# Add withCategories and withAttributes HOCs

This PR:
* Creates `withCategories` and `withAttributes` HOC.
* Makes it so Product Category Control and Product Attribute Control display API errors to the user.
* Refactors error handling in all HOCs so JS errors can be handled too.

### Screenshots
![image](https://user-images.githubusercontent.com/3616980/64109532-0bdc3680-cd80-11e9-85fc-aeb9a2077684.png)

### How to test the changes in this Pull Request:

1. Create a post with a _Products by Category_ and _Products by Attribute_ blocks.
2. In `src/RestApi/Controllers/ProductCategories.php` modify [this line](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/src/RestApi/Controllers/ProductCategories.php#L84) with:
```diff
-if ( ! $taxonomy || ! \taxonomy_exists( $taxonomy ) ) {
+if( true ) {
```
3. In `src/RestApi/Controllers/ProductAttributes.php` modify [this one](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/src/RestApi/Controllers/ProductAttributes.php#L81) with:
```diff
-if ( ! current_user_can( 'edit_posts' ) ) {
+if ( true ) {
```
4. Reload the post and verify errors are displayed.
